### PR TITLE
Style reset password to match design

### PR DESCRIPTION
**Pull request by jryans**
Fixes https://github.com/vector-im/riot-web/issues/8169 in large part. As with the other auth screens, this is just major layout and flow changes. Fields will still be tweaked in future work.

image
image

### CODE REVIEW

File: src/components/structures/auth/ForgotPassword.js [hunk 1]
```diff
@@ -175,95 +203,144 @@ module.exports = React.createClass({
...
+        }
+
+        let yourMatrixAccountText = _t('Your account');
+        try {
+            const parsedHsUrl = new URL(this.state.enteredHsUrl);
+            yourMatrixAccountText = _t('Your account on %(serverName)s', {
+                serverName: parsedHsUrl.hostname,
+            });
+        } catch (e) {
+            // ignore
```
Comment (turt2live) [inline src/components/structures/auth/ForgotPassword.js]: Not sure this is the best option: shouldn't we be informing the user that their entered URL is wrong?

File: src/components/structures/auth/ForgotPassword.js [hunk 2]
```diff
@@ -175,95 +203,144 @@ module.exports = React.createClass({
...
+                        onChange={this.onInputChanged.bind(this, "email")}
+                        placeholder={_t('Email')} autoFocus />
+                </div>
+                <div className="mx_AuthBody_fieldRow">
+                    <input className="mx_Login_field" ref="pass" type="password"
+                        name="reset_password"
+                        value={this.state.password}
+                        onChange={this.onInputChanged.bind(this, "password")}
+                        placeholder={_t('Password')} />
+                    <input className="mx_Login_field" ref="pass" type="password"
```
Comment (turt2live) [inline src/components/structures/auth/ForgotPassword.js]: duplicate `ref`

File: src/components/structures/auth/ForgotPassword.js [hunk 3]
```diff
@@ -175,95 +203,144 @@ module.exports = React.createClass({
...
+            <input className="mx_Login_submit" type="button" onClick={this.onVerify}
+                value={_t('I have verified my email address')} />
+        </div>;
+    },
+
+    renderDone() {
+        return <div>
+            <p>{_t('Your password has been reset')}.</p>
+            <p>{_t('You have been logged out of all devices and will no longer receive push notifications. ' +
+                'To re-enable notifications, sign in again on each device')}.</p>
```
Comment (turt2live) [inline src/components/structures/auth/ForgotPassword.js]: would be nice to have this either be indented to line up with the other line (similar to how line 302 deals with this) or to have them on independent lines (similar to 284).

File: src/components/structures/auth/ForgotPassword.js (hunk 1 above)
Comment (jryans) [reply src/components/structures/auth/ForgotPassword.js]: Ah, good idea. This was borrowed from the other auth screens, but they each have a different path to suggest your HS URL is bad, while this one doesn't. I added basic validation here, and I also filed https://github.com/vector-im/riot-web/issues/8429 to share the work across all auth pages.

File: src/components/structures/auth/ForgotPassword.js (hunk 2 above)
Comment (jryans) [reply src/components/structures/auth/ForgotPassword.js]: (This code was all just moved around, which may be more obvious from the individual commits.) It seems this component hasn't actually used these refs recently / ever, so I'll remove them. Thanks! 😁

File: src/components/structures/auth/ForgotPassword.js (hunk 3 above)
Comment (jryans) [reply src/components/structures/auth/ForgotPassword.js]: Ah yeah, still not sure I've settled on my favorite style for this... Also, most of this code is moved. Anyway, I'll update this bit.

Comment (jryans) [reply src/components/structures/auth/ForgotPassword.js]: These lines also did a strange thing of leaving the last full stop outside the string... 😕 Fixed now.

**Pull request merged**